src/router.js

```diff
@@ -1,41 +1,38 @@
-export function router(options) {
-  return function(emit) {
-    return {
-      state: {
-        router: {}
-      },
-      actions: {
-        router: {
-          set: function(state, actions, data) {
-            return {
-              router: data
-            }
+import { h } from 'hyperapp'
+
+export function router(routes) {
+  return function (app) {
+    return function (props) {
+      return app(enhance(props))
+
+      function enhance(props) {
+        props = Object.assign({ state: {}, actions: {} }, props)
+
+        props.state.router = {}
+        props.actions.router = {
+          set: function (state, actions, data) {
+            return data
           },
-          go: function(state, actions, path) {
+          go: function (state, actions, path) {
             if (location.pathname + location.search !== path) {
-              history.pushState({}, "", path)
-              actions.router.set({
-                path: path
-              })
+              history.pushState({}, '', path)
+              actions.set({ path: path })
             }
           }
         }
-      },
-      events: {
-        load: function(state, actions) {
-          addEventListener("popstate", function() {
+
+        props.view = function (state, actions) {
+          window.onpopstate = function () {
```

Comment:
Can this be an event listener as opposed to a handler? Overriding global state again seems concerning. Also why does this happen in view as opposed to looking at the actions returned by the call to app?

Reply:
unfortunately it wasn't working outside of the view. i prefer to use `addEventListener` outside the view but can't make it working.

Reply:
please see may improved implementation in the latest commit; `addEventListener` working